src/metabase/driver/sparksql.clj

```diff
@@ -0,0 +1,146 @@
+(ns metabase.driver.sparksql
+  (:require [clojure.java.jdbc :as jdbc]
+            (clojure [set :as set]
+                     [string :as s])
+            (honeysql [core :as hsql]
+                      [helpers :as h])
+            [metabase.db.spec :as dbspec]
+            [metabase.driver :as driver]
+            [metabase.driver.generic-sql :as sql]
+            [metabase.driver.hive-like :as hive-like]
+            [metabase.util :as u]
+            [metabase.util.honeysql-extensions :as hx]
+            [metabase.query-processor.util :as qputil]))
+
+(defn sparksql
+  "Create a database specification for a Spark SQL database. Opts should include
+  keys for :db, :user, and :password. You can also optionally set host,
+  port, and additional JDBC connection string options."
+  [{:keys [host port db jdbc-flags]
+    :or {host "localhost", port 10000, db "", jdbc-flags ""}
+    :as opts}]
+  ;; gen-class doesn't support generating static initializers, so we
+  ;; manually register our FixedHiveDriver with java.sql.DriverManager
+  ;; here and make sure it's the only driver returned for jdbc:hive2
+  ;; Class/forName and invokeConstructor is required to make this compile,
+  ;; but it may be possible to solve this with the right project.clj magic
+  (java.sql.DriverManager/registerDriver
+   (clojure.lang.Reflector/invokeConstructor
+    (Class/forName "metabase.driver.FixedHiveDriver")
+    (into-array [])))
+  (loop []
+    (when-let [driver (try
+                        (java.sql.DriverManager/getDriver "jdbc:hive2://localhost:10000")
+                        (catch java.sql.SQLException e
```

Comment:
https://github.com/bbatsov/clojure-style-guide#underscore-for-unused-bindings

Reply:
Done